### PR TITLE
End Pr210

### PR DESCRIPTION
**Pull request by KostyaSha**
This change is [Reviewable](https://reviewable.io/reviews/jenkinsci/github-plugin/221)

### CODE REVIEW

**Comment by KostyaSha** [comment]
@timja ping

File: pom.xml [hunk 1]
```diff
@@ -54,6 +54,7 @@
         <concurrency>1</concurrency>
         <java.level>8</java.level>
         <workflow.version>1.14.2</workflow.version>
+        <configuration-as-code.version>1.12</configuration-as-code.version>
```
Comment (timja) [inline pom.xml]:
```suggestion
        <configuration-as-code.version>1.32</configuration-as-code.version>
```

File: src/test/java/org/jenkinsci/plugins/github/test/GitHubServerConfigMatcher.java [hunk 2]
```diff
@@ -31,4 +84,50 @@ protected String featureValueOf(GitHubServerConfig actual) {
...
+                return valueOrNull(actual, "name");
+            }
+        };
+    }
+
+    private static String valueOrNull(Mapping mapping, String key) {
+        try {
+            return mapping.get(key).asScalar().getValue();
+        } catch (NullPointerException | ConfiguratorException e) {
+//            LOG.error("", e);
```
Comment (timja) [inline src/test/java/org/jenkinsci/plugins/github/test/GitHubServerConfigMatcher.java]: ?

File: src/test/java/org/jenkinsci/plugins/github/test/GitHubServerConfigMatcher.java [hunk 3]
```diff
@@ -31,4 +84,50 @@ protected String featureValueOf(GitHubServerConfig actual) {
...
+            throw new AssertionError(key);
+//            return "UNKNOWN VALUE: " + key;
```
Comment (timja) [inline src/test/java/org/jenkinsci/plugins/github/test/GitHubServerConfigMatcher.java]: ?

File: src/test/java/org/jenkinsci/plugins/github/config/ConfigAsCodeTest.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                        .and(withApiUrl(is("https://api.some.com")))
+                        .and(withCredsId(is("private_cred_id")))
+                        .and(withClientCacheSize(is(40)))
+                        .and(withIsManageHooks(is(false)))
+        ));
+    }
+
+    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
+    public void exportConfiguration() throws Exception {
```
Comment (timja) [inline src/test/java/org/jenkinsci/plugins/github/config/ConfigAsCodeTest.java]: you could probably write a simpler test with a fixture, there's nothing dynamic here so should be straightforward:
https://github.com/jenkinsci/slack-plugin/blob/master/src/test/java/jenkins/plugins/slack/jcasc/ConfigurationAsCodeTest.java#L34

Comment (KostyaSha) [reply src/test/java/org/jenkinsci/plugins/github/config/ConfigAsCodeTest.java]: What is the difference with the source yaml?

File: src/test/java/org/jenkinsci/plugins/github/test/GitHubServerConfigMatcher.java (hunk 2 above)
Comment (KostyaSha) [reply src/test/java/org/jenkinsci/plugins/github/test/GitHubServerConfigMatcher.java]: There is no methods to check existance of the key. Either get once with removal either get NPE. Exception is expected against object that has no key, no need to log, so commented.

File: src/test/java/org/jenkinsci/plugins/github/test/GitHubServerConfigMatcher.java (hunk 3 above)
Comment (KostyaSha) [reply src/test/java/org/jenkinsci/plugins/github/test/GitHubServerConfigMatcher.java]: the same

File: src/test/java/org/jenkinsci/plugins/github/config/ConfigAsCodeTest.java (hunk 4 above)
Comment (timja) [reply src/test/java/org/jenkinsci/plugins/github/config/ConfigAsCodeTest.java]: the source yaml is the same, the expected one doesn't have the attribute that you do a .get on, in the case of slack it's missing `slackNotifier` but has all the fields under it

File: src/test/java/org/jenkinsci/plugins/github/test/GitHubServerConfigMatcher.java (hunk 2 above)
Comment (timja) [reply src/test/java/org/jenkinsci/plugins/github/test/GitHubServerConfigMatcher.java]: more around is this commented out code intended to stay.

File: src/test/java/org/jenkinsci/plugins/github/config/ConfigAsCodeTest.java (hunk 4 above)
Comment (KostyaSha) [reply src/test/java/org/jenkinsci/plugins/github/config/ConfigAsCodeTest.java]: i not sure also btw, final yaml is missing default return settings, is it ok?

Comment (KostyaSha) [reply src/test/java/org/jenkinsci/plugins/github/config/ConfigAsCodeTest.java]: @timja i can merge&release finally this caasc support and i can accept PR for test update if you want to change it

File: src/test/java/org/jenkinsci/plugins/github/test/GitHubServerConfigMatcher.java (hunk 3 above)
Comment (KostyaSha) [reply src/test/java/org/jenkinsci/plugins/github/test/GitHubServerConfigMatcher.java]: cleaned

File: src/test/java/org/jenkinsci/plugins/github/config/ConfigAsCodeTest.java (hunk 4 above)
Comment (timja) [reply src/test/java/org/jenkinsci/plugins/github/config/ConfigAsCodeTest.java]: probably not worth it, it works and is unlikely to ever need to change.

**Comment by KostyaSha** [comment]
Thanks for CasC :) Waiting for read only global options.

**Comment by casz** [comment]
Thanks @KostyaSha for finishing this up 😊

**Comment by JackHowa** [comment]
This seems like it's a problem. https://issues.jenkins-ci.org/browse/JENKINS-60738

**Pull request merged**